Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp

```diff
@@ -1637,6 +1650,82 @@ int CLuaElementDefs::removeElementData(lua_State* luaVM)
     return 1;
 }
 
+int CLuaElementDefs::subscribeElementData(lua_State* luaVM)
+{
+    //  bool subscribeElementData ( element theElement, string key, player thePlayer )
+    CElement* pElement;
+    SString   strKey;
+    CPlayer*  pPlayer;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadUserData(pElement);
+    argStream.ReadString(strKey);
+    argStream.ReadUserData(pPlayer);
+
+    if (!argStream.HasErrors())
+    {
+        CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+        if (pLuaMain)
+        {
+            LogWarningIfPlayerHasNotJoinedYet(luaVM, pElement);
+
+            if (IS_PLAYER(pPlayer))
+            {
+                if (CStaticFunctionDefinitions::SubscribeElementData(pElement, strKey, pPlayer))
+                {
+                    lua_pushboolean(luaVM, true);
+                    return 1;
+                }
+            }
+            else
+                argStream.SetTypeError("player", 1);

```

Comment:
You also don't need IS_PLAYER checks - this is done by CArgStream
```suggestion
            if (CStaticFunctionDefinitions::SubscribeElementData(pElement, strKey, pPlayer))
            {
                lua_pushboolean(luaVM, true);
                return 1;
            }

```